src/win/fs.c

```diff
@@ -1611,26 +1613,356 @@ static void fs__access(uv_fs_t* req) {
+  }
+
+  /* If we get to the end, everything worked out. */
+  SET_REQ_SUCCESS(req);
 }
 
+static void build_explicit_access(EXPLICIT_ACCESS_W* ea, PSID owner, TRUSTEE_TYPE user_type, mode_t mode_triplet) {
+  /*
+   * We map the typical POSIX mode bits r/w/x as the following windows permissions:
+   * - Read: FILE_READ_DATA | FILE_READ_EA | FILE_READ_ATTRIBUTES
+   * - Write: FILE_WRITE_DATA | FILE_WRITE_EA
+   * - Execute: FILE_EXECUTE | FILE_READ_ATTRIBUTES
+   *
+   * We map the "User", "Group", and "Other" sets as the current object owner,
+   * the current object group, and the "Everyone" group, respectively.
+   */
+  ZeroMemory(ea, 2*sizeof(EXPLICIT_ACCESS_W));
+
+  /* 
+   * Initialize two EXLPICIT_ACCESS structures; one to explicitly allow things, the
+   * other to explicitly deny them.  We leave no middle ground for inheritance to mess
+   * things up.
+   */
+  for(int i=0; i<2; ++i) {
+    ea[i].grfAccessPermissions = 0;
+    ea[i].grfInheritance = NO_INHERITANCE;
+    ea[i].Trustee.MultipleTrusteeOperation = NO_MULTIPLE_TRUSTEE;
+    ea[i].Trustee.TrusteeForm = TRUSTEE_IS_SID;
+    ea[i].Trustee.TrusteeType = user_type;
+    ea[i].Trustee.ptstrName = owner;
+  }
+
+  ea[0].grfAccessMode = GRANT_ACCESS;
+  ea[1].grfAccessMode = DENY_ACCESS;
+  
+  if( mode_triplet & 0x1 ) {
+    ea[0].grfAccessPermissions |= FILE_EXECUTE | FILE_READ_ATTRIBUTES;
+  } else {
+    ea[1].grfAccessPermissions |= FILE_EXECUTE | FILE_READ_ATTRIBUTES;
+  }
+  if( mode_triplet & 0x2 ) {
+    ea[0].grfAccessPermissions |= FILE_WRITE_DATA | FILE_WRITE_EA | FILE_WRITE_ATTRIBUTES | FILE_APPEND_DATA;
+  } else {
+    ea[1].grfAccessPermissions |= FILE_WRITE_DATA | FILE_WRITE_EA | FILE_WRITE_ATTRIBUTES | FILE_APPEND_DATA;
+  }
+  if( mode_triplet & 0x4 ) {
+    ea[0].grfAccessPermissions |= FILE_READ_DATA | FILE_READ_EA | FILE_READ_ATTRIBUTES;
+  } else {
+    ea[1].grfAccessPermissions |= FILE_READ_DATA | FILE_READ_EA;
+    /*
+     * Because we have overlap between x and r, we need to not deny FILE_READ_ATTRIBUTES
+     * if we have excecutable permissions.
+     */
+    if (!(mode_triplet & 0x1)) {
+      ea[1].grfAccessPermissions |= FILE_READ_ATTRIBUTES;
+    }
+  }
+}
 
 static void fs__chmod(uv_fs_t* req) {
-  int result = _wchmod(req->file.pathw, req->fs.info.mode);
-  SET_REQ_RESULT(req, result);
+  PACL pOldDACL = NULL, pNewDACL = NULL;
+  PSID psidOwner = NULL, psidGroup = NULL, psidEveryone = NULL, psidNull = NULL, psidCreatorGroup = NULL;
+  PSECURITY_DESCRIPTOR pSD = NULL;
+  PEXPLICIT_ACCESS_W ea = NULL, poldEAs = NULL;
+
+  /* Create well-known SIDs for the NULL SID, the Everyone group and the meta creator group */
+  SID_IDENTIFIER_AUTHORITY SIDAuthWorld = SECURITY_WORLD_SID_AUTHORITY;
+  SID_IDENTIFIER_AUTHORITY SIDAuthNull = SECURITY_NULL_SID_AUTHORITY;
+  SID_IDENTIFIER_AUTHORITY SIDAuthCreator = SECURITY_CREATOR_SID_AUTHORITY;
+  
+  if(!AllocateAndInitializeSid(&SIDAuthWorld, 1, SECURITY_WORLD_RID, 0, 0, 0, 0, 0, 0, 0, &psidEveryone) ||
```

Comment:
```suggestion
  if(!AllocateAndInitializeSid(&SIDAuthWorld, 1, SECURITY_WORLD_RID, 0, 0, 0, 0, 0, 0, 0, &psidEveryone) ||
```

add space (and fix following alignment)